DigitalLearningSolutions.Data/Models/TrackingSystem/DateInformation.cs

```diff
@@ -5,22 +5,66 @@
 
     public class DateInformation
     {
-        public DateInformation(DateTime date, ReportInterval interval)
+        public DateInformation(DateTime startDate, ReportInterval interval)
         {
-            Date = date;
+            StartDate = startDate;
             Interval = interval;
         }
 
         public ReportInterval Interval { get; set; }
-        public DateTime Date { get; set; }
+        public DateTime StartDate { get; set; }
 
         public string GetDateLabel(string format)
         {
-            var quarter = Date.Month / 3 + 1;
+            format = ConvertQuartersInFormatString(format, StartDate);
 
-            format = format.Replace("Q", $"'Quarter' {quarter}").Replace("q", $"Q{quarter}");
+            return StartDate.ToString(format);
+        }
+
+        public string GetDateRangeLabel(string format, DateTime rangeTerminator, bool startRangeFromTerminator)
+        {
+            var dateFromThisObject = startRangeFromTerminator ? GetFinalDate() : StartDate;
+            var formatForDateFromThisObject = ConvertQuartersInFormatString(format, dateFromThisObject);
+
+            var formatForTerminator = ConvertQuartersInFormatString(format, rangeTerminator);
+
+            return startRangeFromTerminator
+                ? rangeTerminator.ToString(formatForTerminator) + " to " +
+                  dateFromThisObject.ToString(formatForDateFromThisObject)
+                : dateFromThisObject.ToString(formatForDateFromThisObject) + " to " +
+                  rangeTerminator.ToString(formatForTerminator);
+        }
+
+        public string GetDateRangeLabel(string format, DateTime startDate, DateTime endDate)
+        {
+            var startFormat = ConvertQuartersInFormatString(format, startDate);
+            var endFormat = ConvertQuartersInFormatString(format, endDate);
 
-            return Date.ToString(format);
+            return startDate.ToString(startFormat) + " to " + endDate.ToString(endFormat);
+        }
+
+        private string ConvertQuartersInFormatString(string format, DateTime date)
+        {
+            var quarter = date.Month / 3 + 1;
+
+            return format.Replace("Q", $"'Quarter' {quarter}").Replace("q", $"Q{quarter}");
+        }

```

Comment:
The DateRangeLabel one I left as dynamic (is that the term here?) because it's an overload of the other method of the same name, which can't be static. It felt appropriate that they both be called in the same way, but I don't know if that seems like a good principle to you.

Reply:
Hm, I still think it makes sense to make it static if it doesn't rely on any object properties, so I'm happy with how it is now.